app/components/inputgroup/inputgroup.js

```diff
@@ -191,6 +196,42 @@ var InputGroup = React.createClass({
     );
   },
 
+  renderSelect: function() {
+    var self = this;
+    var placeholder = this.props.value === this.props.placeholder;
```

Comment:
why not just use the defined `self` from above in all instances where we are still using `this`?